### DISCUSSION TITLE
How to include assets (images, videos, ...) in an npm package?

### DISCUSSION THREAD

**Discussion by iam-markus** [parcel-bundler/parcel#5908] [Q&A] [answered]
**Context**
I am working on an internal npm library that is used for sharing code across projects.
Some of the components use images or videos that is located inside the library's assets folder.

This is how the folder structure of the lib looks like:
```
my-lib
  - assets
     arrow.png
  - src
     ArrowComponent.js
   package.json

```

Then in `ArrowComponent.js`:
```
import arrow from 'url:/assets/arrow.png';

function ArrowComponent() {
  return <img src={arrow} />
}
```

This works perfectly fine, when I run parcel to test if the component works from INSIDE the library.  
Also building to the `dist/` folder is working fine.

**Problem**
However, when I want to use the library in another project parcel's url loader get's confused:
`@parcel/core: Failed to resolve 'url:/assets/arrow.png' from '../../company/my-lib/src/components/ArrowComponent.js'`
So it' looking outside the project folder, when it should be looking inside node_modules.

**Question**
How can I import assets in my-lib, so consuming projects can use them as well?

**Comment by mischnic** [comment]
1. Don't use `import .... from "url:..."` in npm package. Webpack won't understand them. Use `new URL("./my-image", import.meta.url)` instead.
2. `/` resolved relative to the project root (and is also non-standard), so you should use relative paths instead

**Reply by iam-markus** [reply]
1. I assume you meant parcel or babel instead of Webpack?

I replaced the import with this:
`const arrow = new URL("../assets/arrow.png", import.meta.url);`
This compiles, but does not place the `arrow.png` inside the dist folder.

When importing from the project, and compiling I get in the browser:
`Uncaught SyntaxError: Cannot use 'import.meta' outside a module`

package.json:
```
"main": "lib/index.js",
"source": "src/index.js",
...
"targets": {
    "main": {
      "includeNodeModules": {
        "react": false,
        "react-dom": false
      },
      "isLibrary": true
    }
  }
```
I tried adding type: module with no difference in outcome.

**Reply by mischnic** [reply]
You need a rather recent nightly for `new URL(..., import.meta.url)` to work

**Reply by iam-markus** [reply]
Thanks, that works.

**Comment by iam-markus** [comment, marked as answer]
Using nightly.602 does resolve this issue
`const arrow = new URL("../assets/arrow.png", import.meta.url)`
